x-pack/plugins/security_solution/server/endpoint/routes/metadata/support/query_strategies.ts

```diff
@@ -6,60 +6,11 @@
  */
 
 import { SearchResponse } from '@elastic/elasticsearch/api/types';
-import {
-  metadataCurrentIndexPattern,
-  metadataIndexPattern,
-} from '../../../../../common/endpoint/constants';
-import { HostMetadata, MetadataQueryStrategyVersions } from '../../../../../common/endpoint/types';
+import { metadataCurrentIndexPattern } from '../../../../../common/endpoint/constants';
+import { HostMetadata } from '../../../../../common/endpoint/types';
 import { HostListQueryResult, HostQueryResult, MetadataQueryStrategy } from '../../../types';
 
-export function metadataQueryStrategyV1(): MetadataQueryStrategy {
-  return {
-    index: metadataIndexPattern,
-    extraBodyProperties: {
-      collapse: {
-        field: 'agent.id',
-        inner_hits: {
-          name: 'most_recent',
-          size: 1,
-          sort: [{ 'event.created': 'desc' }],
-        },
-      },
-      aggs: {
-        total: {
-          cardinality: {
-            field: 'agent.id',
-          },
-        },
-      },
-    },
-    queryResponseToHostListResult: (
-      searchResponse: SearchResponse<HostMetadata>
-    ): HostListQueryResult => {
-      const response = searchResponse as SearchResponse<HostMetadata>;
-      return {
-        resultLength:
-          ((response?.aggregations?.total as unknown) as { value?: number; relation: string })
-            ?.value || 0,
-        resultList: response.hits.hits
-          .map((hit) => hit.inner_hits?.most_recent.hits.hits)
-          .flatMap((data) => data)
-          .map((entry) => (entry?._source ?? {}) as HostMetadata),
-        queryStrategyVersion: MetadataQueryStrategyVersions.VERSION_1,
-      };
-    },
-    queryResponseToHostResult: (searchResponse: SearchResponse<HostMetadata>): HostQueryResult => {
-      const response = searchResponse as SearchResponse<HostMetadata>;
-      return {
-        resultLength: response.hits.hits.length,
-        result: response.hits.hits.length > 0 ? response.hits.hits[0]._source : undefined,
-        queryStrategyVersion: MetadataQueryStrategyVersions.VERSION_1,
-      };
-    },
-  };
-}
-
-export function metadataQueryStrategyV2(): MetadataQueryStrategy {
+export function getMetadataQueryStrategy(): MetadataQueryStrategy {
```

Comment:
Open to opinions on this:

Do we even need this function anymore? It's returning some constants (transform dest index, `track_total_hits`) and then two functions that map a list response to final response, and the single response to final response.

None of those things are variable anymore (the response transforming was different between the Aggregation strategy vs Transform strategy). Although I guess all this information is used in two places (fetch 1 for "details", and fetch all for list). 

The functions themselves can probably be extracted out to file-level functions, and then in the handlers (list, details) they can just directly call the corresponding fn exported from here, instead of using `getMetadataQueryStrategy().queryResponseToHostListResult(data)`. No reason to have a function returning a constant function now, eh?

The index is also more or less a constant. So that's still not terrible to take out of here. Which just leaves the total hits thing. Which is also only technically relevant for the "list" handler. Since details is just fetching one thing.

IMO this can be axed and refactored. But I'm open to hearing other thoughts here

Reply:
my opinion on this is that your opinion is correct 😄. axed it 👍.